src/extensions/http_api/client.ts

```diff
@@ -0,0 +1,312 @@
+import pathModule from 'node:path'
+import axios, { type AxiosInstance, type AxiosResponse } from 'axios'
+import { wrapper } from 'axios-cookiejar-support'
+import FormData from 'form-data'
+import { Cookie, CookieJar } from 'tough-cookie'
+import type { CookieProperty, RequestBody, RequestHeaders, RequestOptions } from '../../types.js'
+import { isPlainObject, isString } from '../../utils/index.js'
+
+const BODY_TYPE_JSON = 'json'
+const BODY_TYPE_FORM = 'form'
+const BODY_TYPE_MULTIPART = 'form-data'
+const verbsAcceptingBody = ['POST', 'PUT', 'DELETE', 'PATCH']
+const validateStatus = (status: number) => status >= 200 && status <= 302
+
+const axiosInstance = axios.create({
+    validateStatus,
+})
+let cookieInstance: AxiosInstance
+
+const getClient = (cookieJar?: CookieJar): AxiosInstance => {
+    if (cookieJar) {
+        if (!cookieInstance) {
```

Comment:
```suggestion
        if (cookieJar && !cookieInstance) {
```

Reply:
If the case cookieJar, we need to return "cookieJar", if undefined, we set it. The code actual is correct.

In your suggestion, we will never return `cookieInstance` for the case `cookieJar && cookieInstance`